### PR TITLE
Clesaec/tcomp 2004 dataset lookup

### PR DESCRIPTION
**Pull request by clesaec**
https://jira.talendforge.org/browse/TCOMP-2004 : can read dataset content from TCK connectors

### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant

### CODE REVIEW

File: component-api/src/main/java/org/talend/sdk/component/api/component/InputFinder.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+package org.talend.sdk.component.api.component;
+
+import java.io.Serializable;
+import java.util.Iterator;
+import java.util.Map;
+
+import org.talend.sdk.component.api.record.Record;
+
+public interface InputFinder extends Serializable {
```
Comment (undx) [inline component-api/src/main/java/org/talend/sdk/component/api/component/InputFinder.java:24]: We need some explanation/doc...

Comment (undx) [reply component-api/src/main/java/org/talend/sdk/component/api/component/InputFinder.java:24]: is it a service, a builtin' service, is it the right place? (needs doc updating maybe)

**Review by undx — COMMENTED** [review]
Great work Chris 👍🏻 
Seems lacking some tests on possible use-cases
Did you tested on a connector's repo?
See comments as missing purpose and usage...

File: component-api/src/main/java/org/talend/sdk/component/api/component/InputFinder.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    Iterator<Record> find(final String pluginIdentifier, final String name, final int version,
```
[the remark this answers is not in the document]
Comment (clesaec) [reply component-api/src/main/java/org/talend/sdk/component/api/component/InputFinder.java:26]: On Mapper interface, plugin design "plugin" name. Yves told us to use plugin identifier; so here, it's not plugin name but plugin identifier; so i use a name that can't confuse.

[the remark this answers is not in the document]
Comment (undx) [reply component-api/src/main/java/org/talend/sdk/component/api/component/InputFinder.java:26]: ok, could use also `pluginId`...

**Review by undx — APPROVED** [review]
LGTM

**Pull request merged**